integration/v3_grpc_test.go

```diff
@@ -156,12 +156,11 @@ func TestV3HashKV(t *testing.T) {
 	kvc := toGRPC(clus.RandClient()).KV
 	mvc := toGRPC(clus.RandClient()).Maintenance
 
-	for i := 0; i < 10; i++ {
+	for i := 0; i < 100; i++ {
```

Comment:
There should be a backend test that exercises the bug; triggering it at this level is too indirect.

Here's the test I wrote to confirm this patch works as expected:
```go
// TestBackendWritebackForEach checks that partially written / buffered                                                                                      
// data is visited in the same order as fully committed data.                                                                                                
func TestBackendWritebackForEach(t *testing.T) {                                                                                                             
        b, tmpPath := NewTmpBackend(time.Hour, 10000)                                                                                                        
        defer cleanup(b, tmpPath)                                                                                                                            
                                                                                                                                                             
        tx := b.BatchTx()
        tx.Lock()
        tx.UnsafeCreateBucket([]byte("key"))
        for i := 0; i < 5; i++ {
                k := []byte(fmt.Sprintf("%04d", i))                                                                                                          
                tx.UnsafePut([]byte("key"), k, []byte("bar"))                                                                                                
        }
        tx.Unlock()                                                                                                                                          
        
        // writeback                                                                                                                                         
        b.ForceCommit()
        
        tx.Lock()
        tx.UnsafeCreateBucket([]byte("key"))                                                                                                                 
        for i := 5; i < 20; i++ {                                                                                                                            
                k := []byte(fmt.Sprintf("%04d", i))                                                                                                          
                tx.UnsafePut([]byte("key"), k, []byte("bar"))                                                                                                
        }       
        tx.Unlock()
        
        seq := ""                                                                                                                                            
        getSeq := func(k, v []byte) error {
                seq += string(k)
                return nil
        }       
        rtx := b.ReadTx()
        rtx.Lock()      
        rtx.UnsafeForEach([]byte("key"), getSeq)                                                                                                             
        rtx.Unlock() 
                
        partialSeq := seq
                
        seq = ""
        b.ForceCommit()                                                                                                                                      

        tx.Lock()                                                                                                                                            
        tx.UnsafeForEach([]byte("key"), getSeq)                                                                                                              
        tx.Unlock()
        
        if seq != partialSeq {                                                                                                                               
                t.Fatalf("expected %q, got %q", seq, partialSeq)                                                                                             
        }
}
```